src/data/finset/noncomm_prod.lean

```diff
@@ -183,6 +189,23 @@ given a proof that `+` commutes on all elements `f x` for `x ∈ s`."]
 def noncomm_prod (s : finset α) (f : α → β) (comm : ∀ (x ∈ s) (y ∈ s), commute (f x) (f y)) : β :=
 (s.1.map f).noncomm_prod (by simpa [multiset.mem_map, ←finset.mem_def] using comm)
 
+/-- A helper for rewriting the dependent arguments. -/
+@[to_additive "A helper for rewriting the dependent arguments "]
+lemma noncomm_prod_congr
+  {s₁ s₂ : finset α}  {f g : α → β} (h₁ : s₁ = s₂) (h₂ : ∀ (x ∈ s₂), f x = g x)
+  (comm : ∀ (x ∈ s₁) (y ∈ s₁), commute (f x) (f y)) :
+  noncomm_prod s₁ f comm = noncomm_prod s₂ g
+    (λ x hx y hy, h₂ x hx ▸ h₂ y hy ▸ comm x (h₁.symm ▸ hx) y (h₁.symm ▸ hy)) :=
+multiset.noncomm_prod_congr (multiset.map_congr (congr_arg _ h₁) h₂) _,
```

Comment:
If you remove the lemma as I suggest above, this can be proved as
```suggestion
by simp_rw [noncomm_prod, multiset.map_congr (congr_arg _ h₁) h₂]
```